aws/resource_aws_docdb_subnet_group.go

```
@@ -0,0 +1,230 @@
+	_, err := conn.CreateDBSubnetGroup(&createOpts)
+	if err != nil {
+		return fmt.Errorf("Error creating DocDB Subnet Group: %s", err)
+	}
+
+	d.SetId(groupName)
+
+	return resourceAwsDocDBSubnetGroupRead(d, meta)
+}
+
+func resourceAwsDocDBSubnetGroupRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).docdbconn
+
+	describeOpts := docdb.DescribeDBSubnetGroupsInput{
+		DBSubnetGroupName: aws.String(d.Id()),
+	}
+
+	var subnetGroups []*docdb.DBSubnetGroup
+	if err := conn.DescribeDBSubnetGroupsPages(&describeOpts, func(resp *docdb.DescribeDBSubnetGroupsOutput, lastPage bool) bool {
+		for _, v := range resp.DBSubnetGroups {
+			subnetGroups = append(subnetGroups, v)
+		}
+		return !lastPage
+	}); err != nil {
+		if isAWSErr(err, docdb.ErrCodeDBSubnetGroupNotFoundFault, "") {
+			log.Printf("[WARN] DocDB Subnet Group (%s) not found, removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("error reading DocDB Subnet Group (%s) parameters: %s", d.Id(), err)
+	}
+
+	if len(subnetGroups) != 1 ||
+		*subnetGroups[0].DBSubnetGroupName != d.Id() {
+		return fmt.Errorf("Unable to find DocDB Subnet Group: %s, removing from state", d.Id())
+	}
+
+	subnetGroup := subnetGroups[0]
+	d.Set("name", subnetGroup.DBSubnetGroupName)
+	d.Set("description", subnetGroup.DBSubnetGroupDescription)
+	d.Set("arn", subnetGroup.DBSubnetGroupArn)
+
+	subnets := make([]string, 0, len(subnetGroup.Subnets))
+	for _, s := range subnetGroup.Subnets {
+		subnets = append(subnets, aws.StringValue(s.SubnetIdentifier))
+	}
+	if err := d.Set("subnet_ids", subnets); err != nil {
+		return fmt.Errorf("error setting subnet_ids: %s", err)
+	}
+
+	resp, err := conn.ListTagsForResource(&docdb.ListTagsForResourceInput{
+		ResourceName: subnetGroup.DBSubnetGroupArn,
+	})
+
+	if err != nil {
+		log.Printf("[DEBUG] Error retrieving tags for ARN: %s", aws.StringValue(subnetGroup.DBSubnetGroupArn))
```

We should return an error instead of logging a debug message here. 👍

---

Fixed.